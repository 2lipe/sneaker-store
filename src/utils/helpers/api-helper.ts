import axios from 'axios';
import { baseConfig } from '../../config/baseConfig';

export const api = axios.create({
  baseURL: baseConfig.baseApiUrl,
});
