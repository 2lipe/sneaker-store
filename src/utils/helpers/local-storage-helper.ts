/* eslint-disable consistent-return */
import { ISneakers } from '../../models/Interfaces/ISneakers';

export const saveSneakersLocalStorage = (data: ISneakers[]) => {
  if (typeof window !== 'undefined') {
    window.localStorage.setItem('@trustly-store:sneakers', JSON.stringify(data));
  }
};

export const getSneakerLocalStorage = () => {
  if (typeof window !== 'undefined') {
    const sneakersString = window.localStorage.getItem('@trustly-store:sneakers');

    if (sneakersString) {
      const sneakers: ISneakers[] = JSON.parse(sneakersString);

      return sneakers;
    }

    return null;
  }
};
