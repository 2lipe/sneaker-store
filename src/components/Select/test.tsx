import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import { renderWithTheme } from '../../utils/tests/helpers';

import { Select } from '.';

const opt = [
  { value: '38', label: '38' },
  { value: '39', label: '39' },
];

describe('<Select />', () => {
  it('should be render Select correctly', () => {
    const { container } = renderWithTheme(<Select name="sizes" label="size" options={opt} />);

    expect(screen.getByText(/size/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be render Select with other options', () => {
    const { container } = renderWithTheme(<Select data-testid="select" name="sizes" label="size" options={opt} />);

    expect(screen.getByText(/size/i)).toBeInTheDocument();

    fireEvent.change(screen.getByTestId('select'), {
      target: { value: '39' },
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
