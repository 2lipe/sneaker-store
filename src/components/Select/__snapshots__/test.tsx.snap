// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Select /> should be render Select correctly 1'] = `
.c0 label {
  font-size: 1.2rem;
  font-weight: 400;
  color: #8D8D8D;
  margin-right: 0.8rem;
}

.c1 {
  width: 6.9rem;
  height: 2.7rem;
  padding-left: 0.8rem;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.2rem;
  color: #808080;
  background-color: #FFFFFF;
  border: 1px solid #E8E8E8;
  border-radius: 2.5rem;
}

<div
  class="c0"
>
  <label
    for="sizes"
  >
    size
  </label>
  <select
    class="c1"
    id="sizes"
  >
    <option
      value="38"
    >
      38
    </option>
    <option
      value="39"
    >
      39
    </option>
  </select>
</div>
`;

exports['<Select /> should be render Select with other options 1'] = `
.c0 label {
  font-size: 1.2rem;
  font-weight: 400;
  color: #8D8D8D;
  margin-right: 0.8rem;
}

.c1 {
  width: 6.9rem;
  height: 2.7rem;
  padding-left: 0.8rem;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.2rem;
  color: #808080;
  background-color: #FFFFFF;
  border: 1px solid #E8E8E8;
  border-radius: 2.5rem;
}

<div
  class="c0"
>
  <label
    for="sizes"
  >
    size
  </label>
  <select
    class="c1"
    data-testid="select"
    id="sizes"
  >
    <option
      value="38"
    >
      38
    </option>
    <option
      value="39"
    >
      39
    </option>
  </select>
</div>
`;
